### PR TITLE
DOC: Add docstring for Extensionarray interpolate

### PR DESCRIPTION
**Pull request by ammar-qazi**
- [ ]  #58539 

Fixed:

```
pandas.api.extensions.ExtensionArray.interpolate PR01,SA01
```

### CODE REVIEW

File: pandas/core/arrays/base.py [hunk 1]
```diff
@@ -999,24 +999,87 @@ def interpolate(
...
+            is passed to scipy.interpolate.UnivariateSpline. These methods use
+            the numerical values of the index.
+            Both 'polynomial' and 'spline' require that you also specify an
+            order (int), e.g. arr.interpolate(method='polynomial', order=5).
+            * 'krogh', 'piecewise_polynomial', 'spline', 'pchip', 'akima',
+            'cubicspline': Wrappers around the SciPy interpolation methods
+            of similar names. See Notes.
+            * 'from_derivatives': Refers to scipy.interpolate.BPoly.from_derivatives.
+        axis : int
+            Axis to interpolate along. For 1D NumpyExtensionArray, use 0.
```
Comment (mroeschke) [inline pandas/core/arrays/base.py:1025]: This docstring should not mention `NumpyExtensionArray` at all.

File: pandas/core/arrays/base.py [hunk 2]
```diff
@@ -999,24 +999,87 @@ def interpolate(
...
+            Maximum number of consecutive NaNs to fill. Must be greater than 0.
+        limit_direction : {'forward', 'backward', 'both'}
+            Consecutive NaNs will be filled in this direction.
+            * If 'method' is 'pad' or 'ffill', 'limit_direction' must be 'forward'.
+            * If 'method' is 'backfill' or 'bfill', 'limit_direction' must be
+            'backward'.
+            Raises ValueError if limit_direction is 'forward' or 'both' and method
+            is 'backfill' or 'bfill'.
+            Raises ValueError if limit_direction is 'backward' or 'both' and method
+            is 'pad' or 'ffill'.
```
Comment (mroeschke) [inline pandas/core/arrays/base.py:1038]: Could you put this in it's own `Raises` section? Please review the numpydoc documentation for where a `Raises` section should be placed

File: pandas/core/arrays/base.py [hunk 3]
```diff
@@ -999,24 +999,87 @@ def interpolate(
...
+        - All parameters must be specified as keyword arguments.
+        - The 'krogh', 'piecewise_polynomial', 'spline', 'pchip' and 'akima'
+          methods are wrappers around the respective SciPy implementations of
+          similar names. These use the actual numerical values of the index.
+        - For 1D NumpyExtensionArray, use 0 for the `axis` parameter.
 
         Examples
         --------
-        >>> arr = pd.arrays.NumpyExtensionArray(np.array([0, 1, np.nan, 3]))
+        >>> arr = pd.arrays.NumpyExtensionArray(np.array([0, np.nan, 2, np.nan, 4]))
```
Comment (mroeschke) [inline pandas/core/arrays/base.py:1070]: Why did this example change?

File: pandas/core/arrays/base.py (hunk 1 above)
Comment (ammar-qazi) [reply pandas/core/arrays/base.py]: I've resolved this issue. Plus, I've also included example of `FloatingArray` to make the documentation more helpful for users.

File: pandas/core/arrays/base.py (hunk 2 above)
Comment (ammar-qazi) [reply pandas/core/arrays/base.py]: I borrowed these insights from [pandas.DataFrame.interpolate`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.interpolate.html). However, it appears that there's an issue with that documentation as well. 

First, there's no method of ffill, bfill, or backfill. Besides that, the function fails in all cases if you use "pad."

For instance, even this raises a ValueError:

```
import pandas as pd
import numpy as np

# Create a sample DataFrame with NaN values
df = pd.DataFrame({
    'A': [1, 2, np.nan, np.nan, 5, 6]
})

print("Original DataFrame:")
print(df)

# Use interpolate with pad method
df_interpolated = df.interpolate(method='pad')

print("\nInterpolated DataFrame:")
print(df_interpolated)
```

File: pandas/core/arrays/base.py (hunk 3 above)
Comment (ammar-qazi) [reply pandas/core/arrays/base.py]: I reverted the change.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @ammar-qazi

**Pull request merged**